provider/parameter_test.go

```diff
@@ -686,6 +687,259 @@ data "coder_parameter" "region" {
 	}
 }
 
+// TestParameterValidationEnforcement tests various parameter states and the
+// validation enforcement that should be applied to them. The table is described
+// by a markdown table. This is done so that the test cases can be more easily
+// edited and read.
+//
+// Copy and paste the table to https://www.tablesgenerator.com/markdown_tables for easier editing
+//
+//nolint:paralleltest,tparallel // Parameters load values from env vars
+func TestParameterValidationEnforcement(t *testing.T) {
+	// Some interesting observations:
+	// - Validation logic does not apply to the value of 'options'
+	//	- [NumDefInvOpt] So an invalid option can be present and selected, but would fail
+	// - Validation logic does not apply to the default if a value is given
+	//	- [NumIns/DefInv] So the default can be invalid if an input value is valid.
+	//	  The value is therefore not really optional, but it is marked as such.
+	// - [NumInsNotOptsVal | NumsInsNotOpts] values do not need to be in the option set?
+
+	table := strings.TrimSpace(`
```

Comment:
This table is a bit over the upper limit of what I'd consider "comfortable" to read inline. Might be better extracted to its own file. Bonus: it becomes previewable in GH!